optuna/distributions.py

```diff
@@ -342,16 +346,35 @@ def __init__(self, low: int, high: int, step: int = 1) -> None:
             )
 
         if step != 1:
-            warnings.warn(
-                "`step` accepts only `1`, so `step` is replaced with `1`. "
-                "`step` argument is deprecated and will be removed in the future. "
-                "The removal of this feature is currently scheduled for v4.0.0, "
-                "but this schedule is subject to change.",
-                FutureWarning,
-            )
+            self._warn_step()
 
         self.low = low
         self.high = high
+        self._step = step
+
+    def __repr__(self):
+        # type: () -> str
```

Comment:
Could you tell me why you use type comments for `__repr__` and `_asdict`?

Reply:
Aligned with `BaseDistribution`. `IntLogUniformDistribution` as well as this entire module is mixed with modern/comment style type hints, but I aligned the type hints for this class now to modern style. What do you think?

Reply:
Thank you for your update. I think we migrate the type hinting gradually, and we can use the new style if we do not have specific reasons.